### ISSUE TITLE
Declaring lifetimes in Helix class structs?

### ISSUE THREAD

**Issue by dtpowl** [tildeio/helix#145]
I need to do something along these lines:

```
ruby! {
    class Foo {
        struct { bar: Bar<'a> }
     }
}
```

But of course this doesn't work, because the lifetime `'a` hasn't been declared. But I can't figure out where/how to declare it. In a normal Rust struct, I could say `struct Baz<'a> { bar: Bar<'a> }` but the anonymous-struct Helix syntax doesn't give me a place to declare the lifetime. I've tried some different things but they all result in the macro-expansion failing.

Am I missing something? Is this just a missing feature at the moment?

**Comment by chancancode** [comment]
You are correct, this is a missing feature at the moment. There are some pending parser changes that needs to happen, but I'll work on this next.
